scopes/lanes/ui/hooks/lanes-query/lanes-query.ts

```diff
@@ -68,18 +79,76 @@ export function useLanesQuery(viewedLaneId?: string): { lanes?: LanesModel } & O
 }
 
 export function useLaneComponentsQuery(lane: LaneModel): {
-  components?: Array<ComponentModel>;
+  components?: Array<{ model: ComponentModel; descriptor: ComponentDescriptor }>;
 } & Omit<QueryResult<LanesQuery>, 'data'> {
   const { data, ...rest } = useDataQuery(GET_LANE_COMPONENTS, {
     variables: { ids: [lane.name] },
   });
 
-  const components: Array<ComponentModel> = data?.lanes.list[0].components.map((component) =>
-    ComponentModel.from({ ...component, host: data.getHost.id })
-  );
+  const components = data?.lanes.list[0].components.map((component) => {
+    const componentModel = ComponentModel.from({ ...component, host: data.getHost.id });
+    const aspectList = {
+      entries: component?.aspects,
+    };
+
+    const componentDescriptor = ComponentDescriptor.fromObject({ id: componentModel.id.toString(), aspectList });
+    return { model: componentModel, descriptor: componentDescriptor };
+  });
 
   return {
     ...rest,
     components,
   };
 }
+
+const GET_LANE_README_COMPONENT = gql`
+  query LaneReadmeComponent(
+    $ids: [String!]
+    $extensionId: String
+    $logType: String
+    $logOffset: Int
+    $logLimit: Int
+    $logHead: String
+    $logSort: String
+  ) {
+    lanes {
+      id
+      list(ids: $ids) {
+        id
+        remote
+        isMerged
+        readmeComponent {
+          ...componentFields
+        }
+      }
+    }
+    getHost(id: $extensionId) {
+      id
+    }
+  }
+  ${componentFields}
+`;
+
+export function useLaneReadmeQuery(lane: LaneModel): {
```

Comment:
I would possibly separate to a different hook component. lanes-query is becoming quite a big component